components/board/numato/board.go

```diff
@@ -354,12 +356,43 @@ type analog struct {
 	pin string
 }
 
-func (a *analog) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
+// Read returns the analog value with the range and step size in V/bit.
+func (a *analog) Read(ctx context.Context, extra map[string]interface{}) (board.AnalogValue, error) {
 	res, err := a.b.doSendReceive(ctx, fmt.Sprintf("adc read %s", a.pin))
 	if err != nil {
-		return 0, err
+		return board.AnalogValue{}, err
+	}
+	reading, err := strconv.Atoi(res)
+	if err != nil {
+		return board.AnalogValue{}, err
 	}
-	return strconv.Atoi(res)
+	var max float32
+	var stepSize float32
+	switch a.b.productID {
+	case 0x805:
+		// 128 channel usb numato has 12 bit resolution
+		max = 3.3
+		stepSize = max / 4096
+	case 0x802:
+		// 32 channel usb numato has 10 bit resolution
+		max = 3.3
+		stepSize = max / 1024
+	case 0x800:
```

Comment:
Mild preference to sort the cases in the switch statement, so it's easier to find the one you care about.